src/Lachain.Core/RPC/HTTP/HttpService.cs

```diff
@@ -169,20 +187,27 @@ private bool _Handle(HttpListenerContext context)
 
         private bool _CheckAuth(JObject body, HttpListenerContext context, string sig, string bodyPlainText)
         {
-            if (context.Request.IsLocal) return true;
+            if(context.Request.IsLocal) return true;
 
-            if (_privateMethods.Contains(body["method"]!.ToString()))
+            if(_privateMethods.Contains(body["method"]!.ToString()))
             {
-                if (string.IsNullOrEmpty(sig))
+                if(string.IsNullOrEmpty(sig))
                 {
                     return false;
                 }
 
-                var messageBytes = Encoding.UTF8.GetBytes(bodyPlainText);
+                string[] bodyParams = bodyPlainText.Split(";");
```

Comment:
It is a wrong way to get message to sign. You have `body` here,  already parsed,  you can construct message to sign from it directly smth like `var msgToSign = body['method'] + serialize(body['params']) + timestamp`